src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -87,6 +99,58 @@ public void RunApiCompat()
             _dict.Clear();
         }
 
+        private IAssemblySymbol GetAssemblySymbolFromStream(Stream assemblyStream, MetadataInformation assemblyInformation, out bool resolvedReferences)
+        {
+            resolvedReferences = false;
+            HashSet<string> directories = null;
+
+            // In order to enable reference support for baseline suppression we need a better way
+            // to resolve references for the baseline package. Let's not enable it for now.
+            bool shouldResolveReferences = !_isBaselineSuppression && _referenceDirectories != null &&
+                _referenceDirectories.TryGetValue(assemblyInformation.TargetFramework, out directories);
+
+            AssemblySymbolLoader loader = new(resolveAssemblyReferences: shouldResolveReferences);
+            if (shouldResolveReferences)
+            {
+                resolvedReferences = true;
+                loader.AddReferenceSearchDirectories(directories);
+            }
+            else if (!_isBaselineSuppression && _referenceDirectories != null)
+            {
+                _log.LogWarning(
+                    new Suppression()
+                    {
+                        DiagnosticId = ApiCompatibility.DiagnosticIds.SearchDirectoriesNotFoundForTfm,
+                        Target = assemblyInformation.DisplayString
+                    },
+                    ApiCompatibility.DiagnosticIds.SearchDirectoriesNotFoundForTfm,
+                    "Could not find a reference directory in the provided directories for TargetFramework '{0}' when loading '{1}'. For more information see: https://aka.ms/dotnetpackagevalidation",
+                    assemblyInformation.TargetFramework, assemblyInformation.DisplayString);
+            }
+
+            IAssemblySymbol symbol = loader.LoadAssembly(assemblyInformation.AssemblyName, assemblyStream);
+
+            if (loader.HasLoadWarnings(out IEnumerable<AssemblyLoadWarning> warnings))
+            {
+                resolvedReferences = false;
+                foreach (AssemblyLoadWarning warning in warnings)
+                {
+                    _log.LogWarning(
+                        new Suppression()
```

Comment:
Same goes for this warning, should it be message instead?

Reply:
Messages are effectively never observed except for diagnostics, so I don't think we'd use a message.